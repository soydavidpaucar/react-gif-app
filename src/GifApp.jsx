export const GifApp = () => {
  return (
    <>
      <h1>GifApp</h1>
    </>
  );
};
