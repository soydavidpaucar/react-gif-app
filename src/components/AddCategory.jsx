import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState('');
  
  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };
  
  const onSubmit = (event) => {
    event.preventDefault();
    
    if (inputValue.trim().length > 0) {
      onNewCategory(inputValue.trim());
      setInputValue('');
    }
  };
  
  return (
    <form onSubmit={onSubmit}>
      <input type="text"
             className="border border-[#64ffda] rounded px-3 py-2 text-[#ccd6f6] outline-none bg-transparent"
             placeholder="Search gifs..." value={inputValue} onChange={handleInputChange} />
    </form>
  );
};