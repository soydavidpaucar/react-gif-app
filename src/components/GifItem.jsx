export const GifItem = ({title, url}) => {
  return (
    <div className="items-center flex justify-center flex-col bg-white rounded-md mb-[20px] h-[0%] mr-[20px]">
      <img src={url} alt={title} className="w-full rounded-t-md"/>
      <p className="text-[#0a192f]">{title}</p>
    </div>
  );
};