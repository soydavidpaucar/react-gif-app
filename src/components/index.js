export * from './AddCategory.jsx';
export * from './GifGrid.jsx';
export * from './GifItem.jsx';