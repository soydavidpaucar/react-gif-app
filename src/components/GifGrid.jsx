import { useEffect, useState } from 'react';

import { getGifs } from '../helpers/getGifs.js';
import { GifItem } from './GifItem';

export const GifGrid = ({category}) => {
  const [images, setImages] = useState([]);
  
  useEffect(() => {
    getGifs(category).then(setImages);
  }, []);
  
  return (
    <>
      <h3 className="text-[#ccd6f6] text-4xl mb-2">{category}</h3>
      <div className="text-[#8791af] mb-12 flex flex-row flex-wrap justify-center">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};