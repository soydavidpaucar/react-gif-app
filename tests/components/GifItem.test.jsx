import { cleanup, render, screen } from '@testing-library/react';
import { afterEach, describe, expect, it, test } from 'vitest';

import { GifItem } from '../../src/components';

/* Cleaning up the DOM after each test. */
afterEach(cleanup);

describe('test GifItem component', () => {
  const title = 'Test Title';
  const url = 'https://test.url/';
  
  it('should match GifItem component with Snapshot', () => {
    const {container} = render(<GifItem title={title} url={url} />);
    
    expect(container).toMatchSnapshot();
  });
  
  it('should display the title, url and alt attributes', () => {
    render(<GifItem title={title} url={url} />);
    
    const {src, alt} = screen.getByRole('img');
    
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  
  it('should display the title in the component', () => {
    render(<GifItem title={title} url={url} />);
    
    const titleElement = screen.getByText(title).innerHTML;
    expect(titleElement).toBe(title);
  });
});
