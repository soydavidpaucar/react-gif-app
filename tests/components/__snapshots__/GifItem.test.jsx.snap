// Vitest Snapshot v1

exports[`test GifItem component > should match GifItem component with Snapshot 1`] = `
<div>
  <div
    class="items-center flex justify-center flex-col bg-[#ccd6f6] rounded-md mb-[20px] h-[0%] mr-[20px]"
  >
    <img
      alt="Test Title"
      class="w-full rounded-t-md"
      src="https://test.url/"
    />
    <p
      class="text-[#0a192f]"
    >
      Test Title
    </p>
  </div>
</div>
`;
